### PR TITLE
TASK: Refactor absolute namespaces in Flow / Core classes

### PR DESCRIPTION
**Pull request by daniellienert**
Also refactored legacy array syntax.

### CODE REVIEW

File: TYPO3.Flow/Classes/TYPO3/Flow/Core/ApplicationContext.php [hunk 1]
```diff
@@ -49,15 +50,15 @@ class ApplicationContext
...
+     * @var ApplicationContext
      */
     protected $parentContext;
 
     /**
      * Initialize the context object.
      *
      * @param string $contextString
-     * @throws \TYPO3\Flow\Exception if the parent context is none of "Development", "Production" or "Testing"
+     * @throws Exception if the parent context is none of "Development", "Production" or "Testing"
```
Comment (albe) [inline TYPO3.Flow/Classes/TYPO3/Flow/Core/ApplicationContext.php]: `Flow\Exception` or `FlowException` maybe

File: TYPO3.Flow/Classes/TYPO3/Flow/Core/Booting/Scripts.php [hunk 2]
```diff
@@ -149,25 +173,25 @@ public static function initializePackageManagement(Bootstrap $bootstrap)
      *
      * @param Bootstrap $bootstrap
      * @return void
-     * @throws \TYPO3\Flow\Exception
+     * @throws Exception
```
Comment (albe) [inline TYPO3.Flow/Classes/TYPO3/Flow/Core/Booting/Scripts.php]: same as above

File: TYPO3.Flow/Classes/TYPO3/Flow/Core/ApplicationContext.php [hunk 3]
```diff
@@ -71,8 +72,8 @@ public function __construct($contextString)
             $this->parentContext = new ApplicationContext(implode('/', $contextStringParts));
         }
 
-        if (!in_array($this->rootContextString, array('Development', 'Production', 'Testing'))) {
-            throw new \TYPO3\Flow\Exception('The given context "' . $contextString . '" was not valid. Only allowed are Development, Production and Testing, including their sub-contexts', 1335436551);
+        if (!in_array($this->rootContextString, ['Development', 'Production', 'Testing'])) {
+            throw new Exception('The given context "' . $contextString . '" was not valid. Only allowed are Development, Production and Testing, including their sub-contexts', 1335436551);
```
Comment (albe) [inline TYPO3.Flow/Classes/TYPO3/Flow/Core/ApplicationContext.php]: See above.

File: TYPO3.Flow/Classes/TYPO3/Flow/Core/Booting/Scripts.php [hunk 4]
```diff
@@ -149,25 +173,25 @@ public static function initializePackageManagement(Bootstrap $bootstrap)
...
 
-        $environment = new \TYPO3\Flow\Utility\Environment($context);
+        $environment = new Environment($context);
         if (isset($settings['utility']['environment']['temporaryDirectoryBase'])) {
             $defaultTemporaryDirectoryBase = FLOW_PATH_DATA . '/Temporary';
             if (FLOW_PATH_TEMPORARY_BASE !== $defaultTemporaryDirectoryBase) {
-                throw new \TYPO3\Flow\Exception(sprintf('It seems like the PHP default temporary base path has been changed from "%s" to "%s" via the FLOW_PATH_TEMPORARY_BASE environment variable. If that variable is present, the TYPO3.Flow.utility.environment.temporaryDirectoryBase setting must not be specified!', $defaultTemporaryDirectoryBase, FLOW_PATH_TEMPORARY_BASE), 1447707261);
+                throw new Exception(sprintf('It seems like the PHP default temporary base path has been changed from "%s" to "%s" via the FLOW_PATH_TEMPORARY_BASE environment variable. If that variable is present, the TYPO3.Flow.utility.environment.temporaryDirectoryBase setting must not be specified!', $defaultTemporaryDirectoryBase, FLOW_PATH_TEMPORARY_BASE), 1447707261);
```
Comment (albe) [inline TYPO3.Flow/Classes/TYPO3/Flow/Core/Booting/Scripts.php]: See above.

File: TYPO3.Flow/Classes/TYPO3/Flow/Core/Booting/Scripts.php [hunk 5]
```diff
@@ -231,34 +255,34 @@ public static function initializeErrorHandling(Bootstrap $bootstrap)
...
+        $bootstrap->setEarlyInstance(CacheFactory::class, $cacheFactory);
     }
 
     /**
      * Runs the compile step if necessary
      *
      * @param Bootstrap $bootstrap
      * @return void
-     * @throws \TYPO3\Flow\Exception
+     * @throws Exception
```
Comment (albe) [inline TYPO3.Flow/Classes/TYPO3/Flow/Core/Booting/Scripts.php]: And once more

File: TYPO3.Flow/Classes/TYPO3/Flow/Core/Booting/Scripts.php [hunk 6]
```diff
@@ -270,24 +294,24 @@ public static function initializeProxyClasses(Bootstrap $bootstrap)
...
             } else {
-                $phpBinaryPathAndFilename = escapeshellarg(\TYPO3\Flow\Utility\Files::getUnixStylePath($settings['core']['phpBinaryPathAndFilename']));
+                $phpBinaryPathAndFilename = escapeshellarg(Files::getUnixStylePath($settings['core']['phpBinaryPathAndFilename']));
             }
             $command = sprintf('%s -c %s -v', $phpBinaryPathAndFilename, escapeshellarg(php_ini_loaded_file()));
             exec($command, $output, $result);
             if ($result !== 0) {
                 if (!file_exists($phpBinaryPathAndFilename)) {
-                    throw new \TYPO3\Flow\Exception(sprintf('It seems like the PHP binary "%s" cannot be executed by Flow. Set the correct path to the PHP executable in Configuration/Settings.yaml, setting TYPO3.Flow.core.phpBinaryPathAndFilename.', $settings['core']['phpBinaryPathAndFilename']), 1315561483);
+                    throw new Exception(sprintf('It seems like the PHP binary "%s" cannot be executed by Flow. Set the correct path to the PHP executable in Configuration/Settings.yaml, setting TYPO3.Flow.core.phpBinaryPathAndFilename.', $settings['core']['phpBinaryPathAndFilename']), 1315561483);
```
Comment (albe) [inline TYPO3.Flow/Classes/TYPO3/Flow/Core/Booting/Scripts.php]: see above

File: TYPO3.Flow/Classes/TYPO3/Flow/Core/Booting/Scripts.php [hunk 7]
```diff
@@ -270,24 +294,24 @@ public static function initializeProxyClasses(Bootstrap $bootstrap)
...
                 }
-                throw new \TYPO3\Flow\Exception(sprintf('It seems like the PHP binary "%s" cannot be executed by Flow. The command executed was "%s" and returned the following: %s', $settings['core']['phpBinaryPathAndFilename'], $command, PHP_EOL . implode(PHP_EOL, $output)), 1354704332);
+                throw new Exception(sprintf('It seems like the PHP binary "%s" cannot be executed by Flow. The command executed was "%s" and returned the following: %s', $settings['core']['phpBinaryPathAndFilename'], $command, PHP_EOL . implode(PHP_EOL, $output)), 1354704332);
```
Comment (albe) [inline TYPO3.Flow/Classes/TYPO3/Flow/Core/Booting/Scripts.php]: see above

File: TYPO3.Flow/Classes/TYPO3/Flow/Core/Booting/Sequence.php [hunk 8]
```diff
@@ -58,7 +59,7 @@ public function addStep(Step $step, $previousStepIdentifier = 'start')
      *
      * @param string $stepIdentifier
      * @return void
-     * @throws \TYPO3\Flow\Exception
+     * @throws Exception
```
Comment (albe) [inline TYPO3.Flow/Classes/TYPO3/Flow/Core/Booting/Sequence.php]: Aaand here

**Comment by albe** [comment]
:heart: apart from the `Exception`s

**Comment by kitsunet** [comment]
Fixed exceptions, will merge when travis is happy.

**Pull request merged**